Port gelu_backward to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58661 Port silu_backward to structured
* **#58665 Port gelu_backward to structured**
* #58664 Port gelu to structured
* #58663 Port hardshrink to structured
* #58662 Port hardshrink_backward and softshrink_backward to structured



Differential Revision: [D28572527](https://our.internmc.facebook.com/intern/diff/D28572527)

Freey0 committed:
Port gelu_backward to structured

[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/Activation.cpp:
```diff
@@ -158,6 +158,13 @@ TORCH_META_FUNC(gelu) (const Tensor & self) {
+TORCH_META_FUNC(gelu_backward) (
+  const Tensor& grad, const Tensor& self
+) {
+  set_output(0, self.sizes(), {}, self.options().memory_format(LEGACY_CONTIGUOUS_MEMORY_FORMAT), {});
```
same deal as https://github.com/pytorch/pytorch/pull/58662/files#r636167323

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572527).

Freey0 committed:
Update on "Port gelu_backward to structured"


Differential Revision: [D28572527](https://our.internmc.facebook.com/intern/diff/D28572527)

[ghstack-poisoned]

Freey0 committed:
Update on "Port gelu_backward to structured"




Differential Revision: [D28572527](https://our.internmc.facebook.com/intern/diff/D28572527)

[ghstack-poisoned]

Freey0 committed:
Update on "Port gelu_backward to structured"




Differential Revision: [D28572527](https://our.internmc.facebook.com/intern/diff/D28572527)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572527).

ezyang commented:
cuda ci failure real

```
May 22 11:11:08 /var/lib/jenkins/workspace/aten/src/ATen/native/cuda/Activation.cu(507): error: declaration is incompatible with "void at::native::structured_gelu_backward_out_cuda::impl(const at::Tensor &, const at::Tensor &, const at::Tensor &)"
May 22 11:11:08 /var/lib/jenkins/workspace/build/aten/src/ATen/NativeFunctions.h(975): here
May 22 11:11:08 
May 22 11:11:08 1 error detected in the compilation of "/var/lib/jenkins/workspace/aten/src/ATen/native/cuda/Activation.cu".
May 22 11:11:08 CMake Error at torch_cuda_cu_generated_Activation.cu.o.Release.cmake:281 (message):
May 22 11:11:08   Error generating file
May 22 11:11:08   /var/lib/jenkins/workspace/build/caffe2/CMakeFiles/torch_cuda_cu.dir/__/aten/src/ATen/native/cuda/./torch_cuda_cu_generated_Activation.cu.o
May 22 11:11:08 
May 22 11:11:08 
```

Freey0 committed:
Update on "Port gelu_backward to structured"




Differential Revision: [D28572527](https://our.internmc.facebook.com/intern/diff/D28572527)

[ghstack-poisoned]

Freey0 committed:
Update on "Port gelu_backward to structured"




Differential Revision: [D28572527](https://our.internmc.facebook.com/intern/diff/D28572527)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572527).
